agent/eventhandler/task_handler.go

```diff
@@ -15,63 +15,96 @@ package eventhandler
 
 import (
 	"container/list"
+	"sync"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
 )
 
-var handler *taskHandler
+// Maximum number of tasks that may be handled at once by the TaskHandler
+const concurrentEventCalls = 3
 
-func init() {
-	handler = newTaskHandler()
+type eventList struct {
+	// events is a list of *sendableEvents
+	events *list.List
+	// sending will check whether the list is already being handlerd
+	sending bool
+	//eventsListLock locks both the list and sending bool
+	eventListLock sync.Mutex
+}
+
+// TaskHandler encapsulates the the map of a task arn to task and container events
+// associated with said task
+type TaskHandler struct {
+	// submitSemaphore for the number of tasks that may be handled at once
+	submitSemaphore utils.Semaphore
+	// taskToEvents is arn:*eventList map so events may be serialized per task
+	tasksToEvents map[string]*eventList
```

Comment:
I know this is moved code, but I never spotted it there before: this is a memory leak because nothing ever gets removed from this map.  Can you add a TODO here to address that leak?